tornettools/generate.py

```diff
@@ -220,26 +233,66 @@ def __server(args, network, server):
     # tgen starts at the end of shadow's "bootstrap" phase
     process["start_time"] = BOOTSTRAP_LENGTH_SECONDS
 
+    if server['is_hs_server']:
+        # this is an onion service, so tgen should only listen on localhost
+        process["environment"] = "TGENIP=127.0.0.1"
```

Comment:
Just curious, is this strictly necessary? Or does it just ensure that onion service servers don't accidentally get used as public servers?

Reply:
It's not necessary, I just added it to catch errors if we accidentally try connecting to onion services from "public" clients. I'll add an extra comment here.